azure/converters/image_test.go

```diff
@@ -26,6 +27,96 @@ import (
 )
 
 func Test_ImageToPlan(t *testing.T) {
+	cases := []struct {
+		name      string
+		image     *infrav1.Image
+		expect    func(*GomegaWithT, *compute.ImageReference)
+		expectErr bool
+	}{
+		{
+			name: "Should return a specific image reference if image id is specified",
+			image: &infrav1.Image{
+				ID: to.StringPtr("fake-id"),
+				SharedGallery: &infrav1.AzureSharedGalleryImage{
+					SubscriptionID: "fake-sub-id",
+					ResourceGroup:  "fake-rg",
+					Gallery:        "fake-gallery-name",
+					Name:           "fake-image-name",
+					Version:        "v1.0.0",
+				},
+			},
+			expect: func(g *GomegaWithT, result *compute.ImageReference) {
```

Comment:
Wdyt about changing the `expect` part to `expectedResult`? That way, you can directly compare the expected conversion result in w/ something like this `g.Expect(res).To(Equal(tc.expectedResult))` in `t.Run()` and don't need to write it in each test case. Check out [this](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/inboundnatrules/spec_test.go) for an example.

Reply:
Either way should be fine. In all the test case -- we have to any way write the expected `infrav1.AzureSharedGalleryImage` struct. The only difference here is that it wrapped by a function. In fact, I thought about it while writing but it has been written the same in the existing code so did not break the pattern. 
Anyways, if you feel to go with the suggested approach -- I can make that change.

Reply:
I got the `expectedResult` convention from the other tests like in `cluster_test.go`(https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/scope/cluster_test.go). IMO I prefer that method since it checks every field and is easier to understand at a glance. Wdyt @CecileRobertMichon

Reply:
+1 to using `expectedResult` if we're trying to compare the entire struct (in this case looks like we are). In some other tests it's sometimes useful to have an `expect` func so we can customize the checks a bit more by test case and only check a specific desired outcome